submissions/crealime/memory-pair-game/js/main.js

```diff
@@ -0,0 +1,255 @@
+    if (hours.length < 2) hours = 0 + hours
+    let minutes = String(Math.floor(timeDiff / 60))
+    if (minutes.length < 2) minutes = 0 + minutes
+    let seconds = String(Math.floor(timeDiff % 60))
+    if (seconds.length < 2) seconds = 0 + seconds
+    this.diff = `${hours}:${minutes}:${seconds}`
+  }
+  getDiffTime() {
+    return this.diff
+  }
+}
+
+const gameTime = new Time()
+
+function getStartPageTemplate() {
+  return `
+    <div class="start-page">
+      <div class="start-page__title">Select number of&nbsp;pairs</div>
+      <div class="start-page__box">
+        <nav class="start-page__nav">
+          <ul class="start-page__ul">
+            <li class="start-page__li"><a href="#" data-pair="6" class="start-page__link">6</a></li>
+            <li class="start-page__li"><a href="#" data-pair="8" class="start-page__link">8</a></li>
+            <li class="start-page__li"><a href="#" data-pair="10" class="start-page__link">10</a></li>
+            <li class="start-page__li"><a href="#" data-pair="12" class="start-page__link">12</a></li>
+            <li class="start-page__li"><a href="#" data-pair="15" class="start-page__link">15</a></li>
+            <li class="start-page__li"><a href="#" data-pair="18" class="start-page__link">18</a></li>
+            <li class="start-page__li"><a href="#" data-pair="21" class="start-page__link">21</a></li>
+            <li class="start-page__li"><a href="#" data-pair="28" class="start-page__link">28</a></li>
+          </ul>
+        </nav>
+      </div>
+      <label class="start-page__label">
+        <input type="checkbox" class="start-page__checkbox">
+        <span class="start-page__preview">Preview</span>
+      </label>
+    </div>
+  `
+}
+
+function getCardTemplate([url, id]) {
+  return `
+    <div class="game__card flip id-${id}" data-id="${id}">
+      <div class="game__front">
+        <img src="${url}" alt="" class="game__img">
+      </div>
+      <div class="game__back"></div>
+    </div>
+  `
+}
+
+function getResultPageTemplate(time, pairs) {
+  return `
+    <div class="result-page">
+      <div class="result-page__message">Your result is<br>${time}<br>for ${pairs} pairs</div>
+       ${resultMessage}
+      <a href="#" class="result-page__link">Play again</a>
+    </div>
+  `
+}
+
+function getResultMessageTemplate(isBest) {
+  return !isBest
+    ? `<div class="result-page__message_success">It’s your own record</div>`
+    : `<div class="result-page__message_warning">Your record is ${isBest}</div>`
+}
+
+function setTimeInLocaleStorage() {
+  const bestTime = localStorage.getItem(`pairs-${numOfPairs}`) || null;
+  if (!bestTime || bestTime && gameTime.getDiffTime().replace(/\D/g, '') < bestTime.replace(/\D/g, '')) {
+    localStorage.setItem(`pairs-${numOfPairs}`, gameTime.getDiffTime())
+    resultMessage = getResultMessageTemplate(false)
+  }
+  else {
+    resultMessage = getResultMessageTemplate(bestTime)
+  }
+}
+
+function fadeOutFadeIn(foo, num, preview) {
```

Comment:
This is the function that initializes new rendering on the page when fadeOut animation is completed. In general, it can be any function. I renamed it to newRenderFunction.